scripts/tools/metricProgrammable.yml

```diff
@@ -0,0 +1,490 @@
+    - type: "$t_metric_programmable_param_value_info_exp_t*"
+      name: pValueInfo
+      desc: |
+            [out][range(1, *pValueInfoCount)] array of parameter value info.
+            if pValueInfoCount is less than the number of value info available, then driver shall only retrieve that number of value info.
+--- #--------------------------------------------------------------------------
+type: function
+desc: "Create metric handles by applying parameter values on the metric programmable handle."
+version: "1.8"
+class: $tMetric
+name: CreateFromProgrammableExp
+decl: static
+details:
+    - "Multiple parameter values could be used to prepare a metric."
+    - "If parameterCount = 0, the default value of the metric programmable would be used for all parameters"
+    - "Parameter values are updated to pParameterValues in the same order as retrieved using $tMetricProgrammableGetParamInfoExp"
+    - "If multiple metrics are generated, then implementation defined C string will be post-fixed to metric name and description"
+params:
+    - type: $t_metric_programmable_exp_handle_t
+      name: hMetricProgrammable
+      desc: "[in] handle of the metric programmable"
+    - type: $t_metric_programmable_param_value_exp_t*
+      name: pParameterValues
+      desc: "[in] list of parameter values to be set."
+    - type: uint32_t
+      name: parameterCount
+      desc: "[in] Count of parameters to set."
+    - type: const char
+      name: "name[ZET_MAX_METRIC_NAME]"
+      desc: "[in] Metric name to be used."
+    - type: const char
+      name: "description[ZET_MAX_METRIC_DESCRIPTION]"
+      desc: "[in] Metric description to be used."
+    - type: "uint32_t*"
+      name: pMetricHandleCount
+      desc: |
+            [in,out] Pointer to the number of metric handles.
+            if count is zero, then the driver shall update the value with the number of metric handles available for this programmable.
+            if count is greater than the number of metric handles available, then the driver shall update the value with the correct number of metric handles available.
+    - type: "$t_metric_handle_t*"
+      name: phMetricHandles
+      desc: |
+            [in,out][optional][range(0,*pMetricHandleCount)] array of handle of metrics.
+            if count is less than the number of metrics available, then driver shall only retrieve that number of metric handles.
+--- #--------------------------------------------------------------------------
+type: function
+desc: "Create metric group handle."
+version: "1.8"
+class: $tMetricGroup
+name: CreateExp
+decl: static
+details:
+    - "Metric handles from multiple domains may be used for a single metric group."
+    - "Metric handles from different sourceIds (refer $t_metric_programmable_exp_properties_t) are not allowed in a single metric group."
```

Comment:
It seems that both of these comments are more related to when the user adds metrics to a group rather than when creating the group.

Reply:
Move it to add api